snuba/datasets/storages/transactions.py

```diff
@@ -157,7 +157,7 @@ def transactions_migrations(
     dist_table_name="transactions_dist",
     storage_set_key=StorageSetKey.TRANSACTIONS,
     mandatory_conditions=[],
-    prewhere_candidates=["event_id", "project_id"],
+    prewhere_candidates=["event_id", "transaction", "title", "project_id"],
```

Comment:
I think you should add transaction_name 

On the AST the and title field are translated before the prewhere is processed (as expected)

```
ColumnToColumn(None, "transaction", None, "transaction_name"),
ColumnToColumn(None, "message", None, "transaction_name"),
ColumnToColumn(None, "title", None, "transaction_name"),
```
So the prewhere processor would not see a `transaction` field in the query (when processing the AST query). Adding both transaction, title and transaction_name will be safe.